packages/cli-platform-android/src/commands/runAndroid/runOnAllDevices.ts

```diff
@@ -16,14 +16,17 @@ import tryLaunchAppOnDevice from './tryLaunchAppOnDevice';
 import tryLaunchEmulator from './tryLaunchEmulator';
 import tryInstallAppOnDevice from './tryInstallAppOnDevice';
 import {Flags} from '.';
+import {BuildFlags} from '../buildAndroid';
 
 export function getTaskNames(
   appName: string,
-  commands: Array<string>,
+  args: BuildFlags,
+  taskPrefix: 'assemble' | 'install',
 ): Array<string> {
-  return appName
-    ? commands.map((command) => `${appName}:${command}`)
-    : commands;
+  const mode = args.mode || args.variant;
```

Comment:
This condition won't work correct because the `args.mode` has the default value (see [here](https://github.com/react-native-community/cli/blob/main/packages/cli-platform-android/src/commands/buildAndroid/index.ts#L120)), so if we provide the `--variant` flag it'll ignored and the `--mode` will be used :( 

So my suggestion is to the delete the default value there and add here something like this:
`const mode = args.mode || args.variant || 'debug';`